src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-import/rgw-multisite-import.component.ts

```diff
@@ -55,23 +90,76 @@ export class RgwMultisiteImportComponent implements OnInit {
             return this.zoneNames && this.zoneNames.indexOf(zoneName) !== -1;
           })
         ]
-      })
+      }),
+      rgw_frontend_port: new FormControl(null, {
+        validators: [Validators.required, Validators.pattern('^[0-9]*$')]
+      }),
+      placement: new FormControl('hosts'),
+      label: new FormControl(null, [
+        CdValidators.requiredIf({
+          placement: 'label',
+          unmanaged: false
+        })
+      ]),
+      hosts: new FormControl([]),
+      count: new FormControl(null, [CdValidators.number(false)]),
+      unmanaged: new FormControl(false)
     });
   }
 
   onSubmit() {
     const values = this.importTokenForm.value;
-    this.rgwRealmService.importRealmToken(values['realmToken'], values['zoneName']).subscribe(
-      () => {
-        this.notificationService.show(
-          NotificationType.success,
-          $localize`Realm token import successfull`
-        );
-        this.activeModal.close();
-      },
-      () => {
-        this.importTokenForm.setErrors({ cdSubmitButton: true });
+    const serviceSpec: object = {
+      placement: {},
+      unmanaged: values['unmanaged']
+    };
+    if (!values['unmanaged']) {
+      switch (values['placement']) {
+        case 'hosts':
+          if (values['hosts'].length > 0) {
+            serviceSpec['placement']['hosts'] = values['hosts'];
+          }
+          break;
+        case 'label':
+          serviceSpec['placement']['label'] = values['label'];
+          break;
       }
-    );
+      if (_.isNumber(values['count']) && values['count'] > 0) {
+        serviceSpec['placement']['count'] = values['count'];
+      }
+    }
+    this.rgwRealmService
+      .importRealmToken(
+        values['realmToken'],
+        values['zoneName'],
+        values['rgw_frontend_port'],
+        JSON.stringify(serviceSpec)
```

Comment:
Here I have two comments: 

1. Using the ServiceSpec class in this case sounds too generic as it seems in fact what you are using is just the `placement spec` part of it which has its own class: `PlacementSpec`. Please, use this class instead of the generic `ServiceSpec`. 

2. The class PlacementSpec has already support for serialization from string (have a look to `PlacementSpec.from_string`) so I don't see the need to use json. I'd recommend to pass the spec as string, this way you don't need to modify anything from the `rgw/module.py` as it's already prepared to receive the placement as string (the typing is already prepared to enforce a string).

Reply:
Thanks @rkachach , we have the `PlacementSpec` class in the python code. In the frontend, we send the placement spec as an object even when we create an rgw service using the dashboard service form which is defined here - 

`export interface CephServicePlacement {
  count: number;
  placement: string;
  hosts: string[];
  label: string;
}`

I have updated the PR with some changes related to type casting , PTAL